tests/cpp/test_alias.cpp

```diff
@@ -1049,7 +1049,43 @@ TEST_F(AliasTest, SegmentBoundary) {
           HeuristicIs(ScheduleHeuristic::PointWise)));
 }
 
-TEST_F(AliasTest, ReuseBufferAliasAcrossSegments) {
+TEST_F(AliasTest, ReuseBuffer) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  TensorView* in = makeContigTensor(1);
+  TensorView* out = add(in, IrBuilder::create<Val>(1.0));
+  fusion->addInput(in);
+  fusion->addOutput(out);
+  fusion->aliasOutputToInput(out, in, AllocationType::ReuseBuffer);
```

Comment:
Naive question: when an output is an alias of an input, doesn't that mean the output always reuse the same buffer as the input?

Reply:
It used to be -- we only used that to mark in-place updates. I've overloaded the API to mark aliases introduced by `slice`s (which doesn't strictly output the same buffer) and to mark for ATen evaluation. Therefore, a separate `AlloationType::Evaluate`. 

I'm open to ideas to clean this up. In the meantime, I believe we'll be able to explicitly represent buffer allocation and ATen evaluation in host IR, instead of relying on attributes on `Fusion`.